test/WebJobs.Script.Tests/DependencyTests.cs

```diff
@@ -0,0 +1,153 @@
+                if (_excludedList.Contains(fileName))
+                {
+                    continue;
+                }
+
+                RuntimeFile newFile = newAssets.SingleOrDefault(p =>
+                {
+                    return Path.GetFileName(p.Path) == fileName &&
+                        p.FileVersion != oldFile.FileVersion &&
+                        p.AssemblyVersion != oldFile.AssemblyVersion;
+                });
+
+                if (newFile != null)
+                {
+                    sb.AppendLine($"    - {fileName}: {oldFile.AssemblyVersion}/{oldFile.FileVersion} -> {newFile.AssemblyVersion}/{newFile.FileVersion}");
+                    changed.Add(oldFile);
+                    changed.Add(newFile);
+                }
+            }
+
+            sb.AppendLine();
+            sb.AppendLine("  Removed:");
+            foreach (RuntimeFile f in removed.Except(changed))
+            {
+                sb.AppendLine($"    - {Path.GetFileName(f.Path)}: {f.AssemblyVersion}/{f.FileVersion}");
+            }
+            sb.AppendLine();
+            sb.AppendLine("  Added:");
+            foreach (RuntimeFile f in added.Except(changed))
+            {
+                sb.AppendLine($"    - {Path.GetFileName(f.Path)}: {f.AssemblyVersion}/{f.FileVersion}");
+            }
+
+            Assert.True(succeed, sb.ToString());
+        }
+
+        private IEnumerable<RuntimeFile> GetRuntimeFiles(string depsJsonFileName)
+        {
+            using (Stream s = new FileStream(depsJsonFileName, FileMode.Open))
+            {
+                DependencyContext deps = _reader.Read(s);
+
+                return deps.RuntimeLibraries
+                    .SelectMany(l => SelectRuntimeAssemblyGroup(_rids, l.RuntimeAssemblyGroups))
+                    .Where(l => !_excludedList.Contains(Path.GetFileName(l.Path)))
+                    .OrderBy(p => Path.GetFileName(p.Path));
+            }
+        }
+
+        private static IEnumerable<RuntimeFile> SelectRuntimeAssemblyGroup(IEnumerable<string> rids, IReadOnlyList<RuntimeAssetGroup> runtimeAssemblyGroups)
+        {
+            // Attempt to load group for the current RID graph
+            foreach (var rid in rids)
+            {
+                var assemblyGroup = runtimeAssemblyGroups.FirstOrDefault(g => string.Equals(g.Runtime, rid, StringComparison.OrdinalIgnoreCase));
+                if (assemblyGroup != null)
+                {
+                    return assemblyGroup.RuntimeFiles;
+                }
+            }
+
+            // If unsuccessful, load default assets, making sure the path is flattened to reflect deployed files
+            return runtimeAssemblyGroups.GetDefaultRuntimeFileAssets();
+        }
+
+        private class RuntimeFileComparer : IEqualityComparer<RuntimeFile>
+        {
+            public bool Equals([AllowNull] RuntimeFile x, [AllowNull] RuntimeFile y)
+            {
+                return x.AssemblyVersion == y.AssemblyVersion &&
```

Comment:
Add name to this as well? (path should cover, but it would be good to be explicit)

Reply:
`RuntimeFile` doesn't have a name -- just these 3 properties: https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencymodel.runtimefile?view=dotnet-plat-ext-5.0&viewFallbackFrom=netcore-3.1

Reply:
I should have checked. Makes sense.